substrate/frame/asset-conversion/src/swap.rs

```diff
@@ -0,0 +1,210 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Traits and implementations for swap between the various asset classes.
+
+use super::*;
+
+/// Trait for providing methods to swap between the various asset classes.
+pub trait Swap<AccountId> {
```

Comment:
Why not put this into the types file where we have other traits?

Reply:
in my opinion it makes it easier to navigate